### PR TITLE
Add application settings for changing the application name

### PR DESCRIPTION
**Pull request by aeschylus**
demo

### CODE REVIEW

File: spec/controllers/sites_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# is no simpler way to get a handle on the object needed for the example.
+# Message expectations are only used when there is no simpler way to specify
+# that an instance is receiving a specific message.
+
+RSpec.describe SitesController, type: :controller do
+  # This should return the minimal set of attributes required to create a valid
+  # Site. As you add validations to Site, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) do
+    skip("Add a hash of attributes valid for your model")
```
Comment (cbeer) [inline spec/controllers/sites_controller_spec.rb]: Can we fill `valid_attributes` and `invalid_attributes` out? That'll get the scaffolding tests up and running.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -33,4 +34,8 @@ def require_active_account!
     def multitenant?
       Settings.multitenant
     end
+
+    def set_site_i18n_keys
+      I18n.backend.store_translations(:en, sufia: { product_name: Site.instance.application_name })
```
Comment (cbeer) [inline app/controllers/application_controller.rb]: This is hideous; hopefully with https://github.com/projecthydra/sufia/pull/1783, we can now override the `application_name` helper.

File: app/controllers/sites_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+class SitesController < ApplicationController
+  before_action :set_site
```
Comment (cbeer) [inline app/controllers/sites_controller.rb]: We should update this controller to use `load_and_authorize_resource` (from cancan) instead of `set_site`.

**Comment by cbeer** [comment]
It looks like this PR needs to be rebased too.

File: app/controllers/sites_controller.rb (hunk 3 above)
Comment (jcoyne) [inline app/controllers/sites_controller.rb]: Can you make this a ticket or fix now?

File: app/models/site.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class Site < ActiveRecord::Base
+  validates :application_name, presence: true, allow_nil: true
```
Comment (jcoyne) [inline app/models/site.rb]: can you explain what this validator does? How is it simultaneously validating presence and allowing nil?

Comment (cbeer) [inline app/models/site.rb]: My intent was to require users to set an application title, but allow it to be unset initially.

File: app/controllers/sites_controller.rb (hunk 3 above)
Comment (cbeer) [inline app/controllers/sites_controller.rb]: Fixed here.

File: app/models/site.rb (hunk 4 above)
Comment (jcoyne) [inline app/models/site.rb]: I see, so empty string is disallowed?

Comment (cbeer) [inline app/models/site.rb]: Exactly. Both cases are incidentally tested in the models and controller tests.

File: app/controllers/sites_controller.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  load_and_authorize_resource
```
Comment (jcoyne) [inline app/controllers/sites_controller.rb]: The load will never be run, because set_site will first assign `@site`, is that intended?

Comment (cbeer) [inline app/controllers/sites_controller.rb]: I figured there was value in consistency; cancancan will short-circuit when it's already loaded, so it isn't harmful.

Comment (jcoyne) [inline app/controllers/sites_controller.rb]: OK

**Pull request merged**